iceoryx_hoofs/test/moduletests/test_vocabulary_semantic_string.cpp

```diff
@@ -67,6 +74,157 @@ template <>
+template <>
+const std::string TestValues<FileName>::SMALLER_VALID_VALUE{"0.me.too.be.file"};
+template <>
+const std::string TestValues<FileName>::MAX_CAPACITY_VALUE{std::string(platform::IOX_MAX_FILENAME_LENGTH, 'b')};
+template <>
+const std::vector<std::string> TestValues<FileName>::INVALID_CONTENT_ADD_BEGIN{};
+template <>
+const std::vector<std::string> TestValues<FileName>::INVALID_CONTENT_ADD_END{};
+///////////////////
+// END: FileName
+///////////////////
+
+///////////////////
+// START: FilePath
+///////////////////
+template <>
+const uint64_t TestValues<FilePath>::CAPACITY = platform::IOX_MAX_PATH_LENGTH;
+template <>
+const std::vector<std::string> TestValues<FilePath>::VALID_VALUES{{"file"},
+                                                                  {"another_file.bla"},
+                                                                  {"123.456"},
+                                                                  {".hidden_me"},
+                                                                  {"/some/file/path"},
+                                                                  {"./relative/path"},
+                                                                  {"another/../../relative/path"},
+                                                                  {"another/../...bla"},
+                                                                  {"not/yet/another/path/../fuu"}};
+template <>
+const std::vector<std::string> TestValues<FilePath>::INVALID_CHARACTER_VALUES{{"some-!user"},
+                                                                              {"*kasjd"},
+                                                                              {"$_fuuuas"},
+                                                                              {";'1'fuuuu"},
+                                                                              {"so*me/path/to/."},
+                                                                              {"/some/pa)th/to/."},
+                                                                              {"another/relative/pa]th/at/the/end/.."}};
+template <>
+const std::vector<std::string> TestValues<FilePath>::INVALID_CONTENT_VALUES{
+    {""}, {"."}, {".."}, {"stop/with/relative/.."}, "another/relative/part/at/the/end/."};
+template <>
+const std::vector<std::string> TestValues<FilePath>::TOO_LONG_CONTENT_VALUES{
+    std::string(platform::IOX_MAX_PATH_LENGTH + 2, 'a')};
+template <>
+const std::string TestValues<FilePath>::GREATER_VALID_VALUE{"9-i-am-a-file"};
+template <>
+const std::string TestValues<FilePath>::SMALLER_VALID_VALUE{"0.me.too.be.file"};
+template <>
+const std::string TestValues<FilePath>::MAX_CAPACITY_VALUE{std::string(platform::IOX_MAX_PATH_LENGTH, 'b')};
+template <>
+const std::vector<std::string> TestValues<FilePath>::INVALID_CONTENT_ADD_BEGIN{};
+template <>
+const std::vector<std::string> TestValues<FilePath>::INVALID_CONTENT_ADD_END{};
```

Comment:
What about a separator at the end?

Reply:
The idea of those variables is that one adds valid content at the end which then makes the result invalid (valid + valid = invalid).
But every content which ends with `/` is invalid so we cannot apply it here.